nixos/modules/system/boot/luksroot.nix

```diff
@@ -148,6 +148,10 @@ let
            + optionalString dev.bypassWorkqueues " --perf-no_read_workqueue --perf-no_write_workqueue"
            + optionalString (dev.header != null) " --header=${dev.header}";
     cschange = "cryptsetup luksChangeKey ${dev.device} ${optionalString (dev.header != null) "--header=${dev.header}"}";
+    fido2luksCredentials = if (dev.fido2.credential != null) then
```

Comment:
you could concat credential and credentials to allow both options at the same time.
If credentials is of type list of string then following expression would allow both:
`fido2luksCredentials = dev.fido2.credentials ++ optional (dev.fido2.credential != null) dev.fido2.credential;`
If fido2luksCredentials is a list of strings, you can easy check for empty list later on.

Reply:
Good point. This made it a lot nicer :)